### PR TITLE
[ARM64] Align Compiler::lvaStressLclFldCB padding

### LINKED ISSUE #15259: JITStress=1 Creates unaligned pointers
With `JITStress=1` or  `JITStress=2`
`CoreMangLib/cti/system/threading/interlocked/InterlockedIncrement1/InterlockedIncrement1` is failing with a stack overflow exception on arm64.

JIT stress is causing the reference to the `COMInterlocked` `location` to be unaligned.  It turns this 
```
               [000039] --C-G-------              /--*  CALL      int    System.Threading.Interlocked.Increment
               [000038] L---G-------              |     \--*  ADDR      byref 
               [000037] ------------              |        \--*  LCL_VAR   blk   (AX) V02 loc1         
``` 
into
```
               [000039] --C-G-------              /--*  CALL      int    System.Threading.Interlocked.Increment
               [000125] ------------              |  |  /--*  CNS_INT   int    2
               [000126] ----G------- arg0         |  \--*  ADD       byref 
               [000038] L---G-------              |     \--*  ADDR      byref 
               [000037] ------------              |        \--*  LCL_VAR   blk   (AX) V02 loc1         
```

The result is that `FastInterlocked*` is throwing a unaligned exception which is a `SIGSEGV` in Ubuntu.

PAL `sigsegv_handler` is turning this into a Stack overflow exception because it occurs near the sp.

I believe the `COMInterlocked` atomic operations should require aligned pointers.  Therefore, my recommendation is that `comutilnative.cpp` checks location alignment and calls `FCThrow(kDataMisalignedException)` for unaligned pointers on all platforms.  

@dotnet/jit-contrib @dotnet/arm64-contrib @jkotas @janvorli Please advise

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #15259 for ARM64

@BruceForstall PTAL

@dotnet/jit-contrib @dotnet/arm64-contrib FYI

### CODE REVIEW

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -1732,11 +1732,13 @@ unsigned CodeGenInterface::InferStructOpSizeAlign(GenTreePtr op, unsigned* align
         LclVarDsc* varDsc = compiler->lvaTable + varNum;
         assert(varDsc->lvType == TYP_STRUCT);
         opSize = varDsc->lvSize();
+#ifndef _TARGET_64BIT_
         if (varDsc->lvStructDoubleAlign)
         {
             alignment = TARGET_POINTER_SIZE * 2;
         }
         else
+#endif
```
Comment (briansull) [inline src/jit/codegencommon.cpp]: Add trailing comment 
`// _TARGET_64BIT_`

Comment (BruceForstall) [reply src/jit/codegencommon.cpp]: Technically, by CC, the comment on endif should match the condition of the preceding section, so
```
// !_TARGET_64BIT_
```
(we generally elide the "defined" in the comments...)

Comment (sdmaclea) [reply src/jit/codegencommon.cpp]: Done

**Comment by sdmaclea** [comment]
@BruceForstall Added casts to fix Windows compile warnings

**Comment by BruceForstall** [comment]
@sdmaclea Looks like a couple more
```
d:\j\workspace\checked_windo---2b2fae86\src\jit\lclvars.cpp(7307): error C2220: warning treated as error - no 'object' file generated [D:\j\workspace\checked_windo---2b2fae86\bin\obj\Windows_NT.x64.Checked\src\jit\protononjit\protononjit.vcxproj]
11:59:53 d:\j\workspace\checked_windo---2b2fae86\src\jit\lclvars.cpp(7307): warning C4267: '=': conversion from 'size_t' to 'unsigned int', possible loss of data [D:\j\workspace\checked_windo---2b2fae86\bin\obj\Windows_NT.x64.Checked\src\jit\protononjit\protononjit.vcxproj]
11:59:53 d:\j\workspace\checked_windo---2b2fae86\src\jit\lclvars.cpp(7308): warning C4267: '=': conversion from 'size_t' to 'unsigned int', possible loss of data [D:\j\workspace\checked_windo---2b2fae86\bin\obj\Windows_NT.x64.Checked\src\jit\protononjit\protononjit.vcxproj]
```

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT armlb Cross Checked Innerloop Build and Test
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test

**Comment by sdmaclea** [comment]
@BruceForstall I believe I fixed all your requested changes.

**Comment by sdmaclea** [comment]
I also believe this can be safely merged.  Ignoring the armlb failure (failing on PRs except #15294)

**Pull request merged**